### PR TITLE
Update branch references

### LINKED ISSUE #1257: Rename master branch to main
see https://github.com/github/renaming/blob/main/README.md

### PR DESCRIPTION
**Pull request by ecdye**
Fixes #1257

Signed-off-by: Ethan Dye <mrtops03@gmail.com>

### CODE REVIEW

**Comment by ecdye** [comment]
@mstormi 

Here we go, if this looks right, then we merge, forward master to stable, update openHAB3 fork from master, and create a main branch and make it the default branch. Does that sound correct?

File: build.bash [hunk 1]
```diff
@@ -2,7 +2,7 @@
 set -e
 
 ####################################################################
-#### dummy: changed this line 18 times to force another image build
+#### dummy: changed this line 19 times to force another image build
```
Comment (mstormi) [inline build.bash:5]: BTW we often don't need this. There's a 'build' action in Github.

File: docs/openhabian.md [hunk 2]
```diff
@@ -299,7 +299,7 @@ This function will rename the rules files so they get ignored by the starting op
 Starting with its general release, openHABian will install **openHAB 3** by default.
 There's some big changes also to openHABian such as to install Java 11 and to use changed file and directory names.
 Most directory names `... /openhab2/ ...` will become `... /openhab/ ...` (NOTE: not `openhab3`) plus there's changes in a number of places, often subtle ones like the name of Samba export shares to change.
-As openHABian users will be running openHAB 2.X by the time 3.0 gets released, `openhabian-config` offers to migrate the openHABian environment and install openHAB3 for you.
+For openHABian users running openHAB 2.X, `openhabian-config` offers to migrate the openHABian environment and install openHAB3 for you.
 Menu option 42 will do the upgrade. Be aware that it isn't the universal answer to all your migration needs - there is ONLY an openHAB upgrade path. You cannot downgrade from OH3 to OH2.
```
Comment (mstormi) [inline docs/openhabian.md:303]: (off the record) did you get this one ?

File: docs/openhabian.md [hunk 3]
```diff
@@ -478,7 +478,7 @@ If you are not able to access your system via the openHAB dashboard or SSH after
 <a id="switch-openhab-branch"></a>
 #### Can I switch openHAB 2 and 3 via openHABian branches ?
 openHABian installs the latest stable build of openHAB.
-The standard openHABian `stable` and `master` branches will install openHAB version 2 and the `openHAB3` branch will install the new openHAB version 3.
+The standard openHABian `openHAB3` and `main` branches will install the new openHAB version 3 and the old `stable` and `master` branches will install the old openHAB version 2.
```
Comment (mstormi) [inline docs/openhabian.md:481]: that would then only be 'stable' no more 'master', right ?

File: functions/openhabian.bash [hunk 4]
```diff
@@ -149,55 +149,65 @@ openhabian_update_check() {
...
+      if [[ $current == "main" ]] || [[ $current == "openHAB3" ]]; then
+          introText="You are currently using the \"${current}\" openHABian environment version.\\n\\nThe very latest openHAB 3 version is called \"main\".\\nThis is providing you with the latest (openHAB3 !) features but less people have tested it so it is a little more likely that you run into errors.\\nYou can step back a little and switch to use the stable version now called \"openHAB3\".\\nYou can switch at any time by selecting this menu option again or by setting the 'clonebranch=' parameter in '/etc/openhabian.conf'.\\n"
       else
           introText="You are currently using an unknown branch of openHABian.\\nThis may be a test version or an error, if so please report on Github (remember to provide a debug log - see debug guide)."
       fi
     fi
 
     if [[ $current == "stable" ]]; then
-      stable="ON"
+      main="OFF"
```
Comment (mstormi) [inline functions/openhabian.bash:180]: Compact this code by setting all to "off" first then only change the variable to represent the "current" branch

**Review by mstormi — APPROVED** [review]
LGTM except for that open question on "master"'s fate

File: build.bash (hunk 1 above)
Comment (ecdye) [reply build.bash:5]: I know, I actually only changed that so I could issue a new commit and rebase it to master.

File: docs/openhabian.md (hunk 2 above)
Comment (ecdye) [reply docs/openhabian.md:303]: For the most part, yes. Not the most elegant phrasing but it works.

File: docs/openhabian.md (hunk 3 above)
Comment (ecdye) [reply docs/openhabian.md:481]: Hmm, I never thought about that. I think we keep master for now and remove it after a little while so we dont break anyone's repositories at that time I will remove it.

File: functions/openhabian.bash (hunk 4 above)
Comment (ecdye) [reply functions/openhabian.bash:180]: That makes sense. This is why we review to fix our stupid mistakes.

File: docs/openhabian.md (hunk 3 above)
Comment (mstormi) [reply docs/openhabian.md:481]: Ok. I think you should add an entry to the NEWS page explaining the new branch structure, asking users of master to migrate over to 'stable'.

Comment (ecdye) [reply docs/openhabian.md:481]: Will do later today. So far the transition seems to have gone smoothly.

File: docs/openhabian.md (hunk 2 above)
Comment (mstormi) [reply docs/openhabian.md:303]: I was referring to the allusion to the Hitch Hiker's Guide ...

**Comment by mstormi** [comment]
Now that we have main I'd think we should release 1.6.3, i.e. tag, build and release another image.
Anything else you think that would need to be in there ?

**Comment by ecdye** [comment]
Not really, although I'm not sure that is necessary because 1.6.2 is on `openHAB3` which is the equivalent of `master`'s `stable`.

**Comment by mstormi** [comment]
true it isn't necessary but as users are unaware of the inner workings it is highly confusing to them to have versions AND branches.
Many don't even know the difference openHAB vs openHABian, I have to correct about 1 posting a week to proclaim some problem in "openhabian 3.0".
So ok, will tag the current release 1.6.3. and upload the latest "main" build image, too

**Comment by mstormi** [comment]
Was just trying to but there's some connectivity problems. Or is it sign of something else ?

```+ introText='You are currently the latest ("openHAB3") openHABian environment version.\nYou will need to be on here (openHAB3) if you want to run openHAB version 3.\n\nYou can switch environments at any time by selecting this menu option again or by setting the '\''clonebranch='\'' parameter in '\''/etc/openhabian.conf'\''.\n\nATTENTION: If you want to upgrade to openHAB 3 you should select "Cancel" now and select option 03 from the main menu instead.'
+ [[ openHAB3 == \s\t\a\b\l\e ]]
+ [[ openHAB3 == \m\a\s\t\e\r ]]
+ [[ openHAB3 == \o\p\e\n\H\A\B\3 ]]
+ stable=OFF
+ master=OFF
+ openHAB3=ON
+ own=no
+ [[ no == \n\o ]]
++ whiptail --title 'openHABian version' --radiolist 'You are currently the latest ("openHAB3") openHABian environment version.\nYou will need to be on here (openHAB3) if you want to run openHAB version 3.\n\nYou can switch environments at any time by selecting this menu option again or by setting the '\''clonebranch='\'' parameter in '\''/etc/openhabian.conf'\''.\n\nATTENTION: If you want to upgrade to openHAB 3 you should select "Cancel" now and select option 03 from the main menu instead.' 14 100 3 stable 'recommended standard version of openHABian' OFF master 'very latest version of openHABian' OFF openHAB3 'openHAB 3.0' ON
+ selection=openHAB3
+ read -r -t 1 -n 1 key
+ [[ -n '' ]]
+ branch=openHAB3
+ sed -i 's|^clonebranch=.*$|clonebranch=openHAB3|g' ''
sed:  kann nicht gelesen werden: Datei oder Verzeichnis nicht gefunden
+ echo 'FAILED (configure clonebranch)'
FAILED (configure clonebranch)
+ exit 1
[11:50:10] root@devpi:/home/openhabian# grep clonebranch /etc/openhabian.conf
clonebranch=openHAB3
[11:50:43] root@devpi:/home/openhabian#
```

**Comment by ecdye** [comment]
Probably just a temporary outage. I have no issues

**Pull request merged**